[IMP] MRP: Basic subcontracting rewrite (#17545)
odoo/documentation: content/applications/inventory_and_mrp/manufacturing/subcontracting/subcontracting_basic.rst

.. note::
   Since contractors are only responsible for purchasing and receiving the final product, they do
   not need to additionally configure manufacturing lead times on a |BoM|. Instead, provide only a
   single *Delivery Lead Time* on the vendor pricelist that factors in both the subcontractor's
   manufacturing and delivery time.

.. _manufacturing/workflows/subcontracting_basic/bom-config:

Configure BoM
-------------

After specifying the vendor, configure a subcontracting-type |BoM| for the product. Click the
:guilabel:`Bill of Materials` smart button on the product form. Then, select the desired |BoM| or
create a new one.

.. tip::
   Alternatively, navigate to :menuselection:`Manufacturing app --> Products --> Bills of
   Materials`, and select the |BoM| for the subcontracted product.

.. image:: subcontracting_basic/basic-sbc-bom.png
   :alt: Bill of Materials configuration for subcontracting in Odoo.

In the :guilabel:`BoM Type` field, select :guilabel:`Subcontracting`. In the resulting
:guilabel:`Subcontractors` field, add one or more subcontractors.

Because the components and manufacturing are both handled by the subcontractor, there is no need to

> list any components in the :guilabel:`Components` tab of the |BoM|.

Reviewer: This image could be cropped to just the **BoM Type** and **Subcontractors** fields, I think. If you keep the full image, maybe move it below line 71.

.. _subcontracting_basic/workflow:

Basic subcontracting workflow
=============================

.. image:: subcontracting_basic/subcontracting-flow.png
   :alt: Basic subcontracting workflow diagram in Odoo.

The basic subcontracting workflow begins with the contracting company :ref:`creating a PO

> <subcontracting_basic/workflow/create_po>` to purchase the product from the subcontractor (1).

Reviewer: ```suggestion
The basic subcontracting workflow begins by :ref:`creating a PO
<subcontracting_basic/workflow/create_po>` to purchase the product from the subcontractor (1).
```

I realize that this was probably an attempt to get around the "You must create...", but it comes off a bit confusing. I believe by "the contracting company" you mean the company that's reading these instructions/selling the product, so I feel we can just launch them into the process.

Author: good suggestion, thanks

> The contractor then confirms the |PO|, which creates a receipt order to receive the final product

Reviewer 2: nitpick: i haven't really seen the inventory experts call these "receipt orders", but rather just "receipts".

However, receipt could imply the form, and it can be confusing if you're trying to refer to the transfer itself. 

Do you think this added context clarifies things or is more confusing? ☠️ I'll let you decide!

```suggestion
The contractor (YourCompany) then confirms the |PO|, which creates a receipt to transfer the final product
```

Author: good suggestion, thanks. going with 'receipt' to standardize the terminology.

(2). The subcontractor manufactures the product and delivers it back to the contractor when done.

> Once the product has been produced and received, the contractor :ref:`validates the receipt order

Reviewer 2: ```suggestion
Once the product has been produced and received, the contractor :ref:`validates the receipt
```

suggesting to remove order here since it's clear you're talking about the form!

<subcontracting_basic/workflow/validate_receipt>` (5) to trigger :ref:`inventory moves
<subcontracting_basic/workflow/track-inventory>` from the subcontractor to the contractor's stock

> (3, 4).

Reviewer: ```suggestion
Once the product has been produced and received, the contractor :ref:`validates the receipt order
<subcontracting_basic/workflow/validate_receipt>` (5) to trigger :ref:`inventory moves
<subcontracting_basic/workflow/track-inventory>` from the subcontractor to the company's stock
(3, 4).
```

--> Purchase Orders`, and click :guilabel:`New`.

Begin filling out the |PO| by selecting a subcontractor from the :guilabel:`Vendor` drop-down menu.
In the :guilabel:`Products` tab, click :guilabel:`Add a product` to create a new product line.
Select the subcontracted product in the :guilabel:`Product` field, and enter the quantity in the
:guilabel:`Quantity` field.

After adding the product, the :guilabel:`Expected Arrival` field is updated with the finished
product's expected delivery date, as configured earlier with the vendor's *Delivery Lead Time*.

Finally, click :guilabel:`Confirm Order` to confirm the |PO|. A receipt is automatically created,
and a :guilabel:`Receipt` smart button appears at the top of the form.

.. _subcontracting_basic/workflow/validate_receipt:

Process receipt
---------------

After the order is confirmed, the subcontractor manufactures the product and delivers the finished
good back to the contracting company.

To receive the finished product from the subcontractor, click the :guilabel:`Receive Products`
button on the |PO|. Alternatively, click the :guilabel:`Receipt` smart button at the top of the

> page, then click :guilabel:`Validate` to enter the incoming shipment into inventory.

Reviewer: ```suggestion
To receive the finished product from the subcontractor, click the :guilabel:`Receive Products`
button on the |PO|, or click the :guilabel:`Receipt` smart button at the top of the
page. Then, click :guilabel:`Validate` to enter the incoming shipment into inventory.
```

I would lump the two ways of getting there together. Because both require the **Validate** button to be clicked, leave it for last in its own sentence.

.. note::
   If :doc:`multi-step inventory flows <../../inventory/shipping_receiving/daily_operations>` are
   enabled, additional transfers must be validated to enter the incoming product into stock.

.. _subcontracting_basic/workflow/track-inventory:

Track inventory moves
---------------------

After validating a receipt, Odoo automatically generates inventory moves to track the movement of
subcontracted products between locations. To view these inventory moves, navigate to
:menuselection:`Inventory app --> Reporting --> Moves History`.

> .. image:: subcontracting_basic/basic-sbc-moves.png

Reviewer: You could possibly make this a bit easier to read by resizing the window you're taking it in and trying to get the table as condensed as you can without compromising the column titles. Because you're taking a screenshot of the entire page, it forces this screenshot to be a lot smaller. There might be a bit of tinkering you can do to make it larger even after resizing.

I would also move this after line 143 (the next paragraph).

   :alt: Moves History page to see subcontracting moves in Odoo.

To track inventory movement in subcontracting, Odoo sends any product components to a dedicated
*Subcontracting Location*. A :ref:`virtual location <inventory/warehouses_storage/location-type>`
called *Production* then consumes the components and produces the finished good. Once produced, the
good then moves back to the *Subcontracting Location* before finally entering the contractor's stock
when the receipt is validated.

.. note::

>    Since no components are sent in basic subcontracting, there is no movement from the

Reviewer: ```suggestion
   Because no components are sent in basic subcontracting, there is no movement from the
```

This is a nit, but it was something I learned a long time ago when I used to use "since" a lot: "Since" implies a period of time. "Because" implies that there is a "cause" for something happening. Because this sentence is explaining *why* something happens, you should use "because."

https://www.grammarly.com/commonly-confused-words/because-vs-since
